test/jdk/java/net/httpclient/ReferenceTracker.java

```diff
@@ -89,10 +92,81 @@ public AssertionError check(long graceDelayMs) {
                 "outstanding operations", true);
     }
 
+    private static String toString(ThreadInfo info) {
```

Comment:
Should we perhaps add a comment to this method explaining why we are duplicating the implementation of `ThreadInfo` here?

I can't find in commit logs or the documentation of the `ThreadInfo` class on why its `toString()` implementation just outputs only 8 stacktrace elements. Do you think we should remove that restriction or document it in a separate issue?

Reply:
Good idea to add a comment. I have also wondered if we should add a new method to ThreadInfo that would take a "maxdepth" integer. I don't know why the output is limited to 8 element. Maybe we should log an enhancement request and let the maintainers of ThreadInfo decide if they want to remove the limitation or provide a new method, or do nothing.